src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -29,42 +30,42 @@ public Range(int first, uint length)
                 throw new ArgumentOutOfRangeException(nameof(first));
             
             First = first;
-            Last = (int)(first + length);
+            End = (int)(first + length);
 
-            if (Last < First) throw new ArgumentOutOfRangeException(nameof(length));
+            if (End < First) throw new ArgumentOutOfRangeException(nameof(length));
         }
 
-        private Range(int first, int last)
+        private Range(int first, int end)
         {
             First = first;
-            Last = last;
+            End = end;
         }
 
-        public bool IsBound => First != UnboundedFirst && Last != UnboundedLast;
+        public bool IsBound => First != UnboundedFirst && End != UnboundedEnd;
```

Comment:
Why can't First == UnboundedFirst? I thought only End is exclusive.

Reply:
The property returns true if both of the ends are bound. Otherwise false. The property can be used to see if some operations on the range would work, e.g. GetEnumerator works only for bound ranges. "Unbound" means "I don't know/care what the value is". It does not make sense to enumerate such ranges.

Reply:
I see. So, you can't enumerate from [int.MinValue..int.MinValue + 5] similar to int.MaxValue?

int.MaxValue being the sentinel for End makes sense given End is exclusive (for example [0..int.MaxValue] works fine. Are we treating the lower bound the same way?